### PR TITLE
ce/ stable and performance fix for show course

### PR DESCRIPTION
**Pull request by CeEv**
# Checkliste

- Bis die komplette Checkliste abgearbeitet ist muss das PR-Label WIP gesetzt sein

## Mehr
Weitere Informationen zur DoD [hier im Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: controllers/courses.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ const logger = require('../helpers/logger');
 
 const router = express.Router();
 
+const { CALENDAR_SERVICE_ENABLED, HOST } = process.env;
```
Comment (tofixx) [inline controllers/courses.js]: This is a feature, we use `FEATURE_...` named ENV's in other situations, why not here?

File: controllers/courses.js [hunk 2]
```diff
@@ -620,99 +622,122 @@ router.get('/:courseId/', (req, res, next) => {
 	];
 
 	// ########################### start requests to new Editor #########################
+	let editorBackendIsAlive = true;
 	if (EDITOR_URL) {
```
Comment (tofixx) [inline controllers/courses.js]: This sounds like a feature, why translating the existence of an url into a feature flag?
Instead I would use a feature flag and if enabled, the url (and maybe other values as well) should be defined.

File: controllers/courses.js [hunk 3]
```diff
@@ -620,99 +622,122 @@ router.get('/:courseId/', (req, res, next) => {
...
-		promises.push(apiEditor(req).get(`course/${req.params.courseId}/lessons`));
+		promises.push(apiEditor(req)
+			.get(`course/${req.params.courseId}/lessons`)
+			.catch((err) => {
+				logger.warn('Can not fetch new editor lessons.', err);
```
Comment (tofixx) [inline controllers/courses.js]: this is an error, isn't it?
```suggestion
				logger.error('Can not fetch new editor lessons.', err);
```

File: controllers/courses.js [hunk 4]
```diff
@@ -620,99 +622,122 @@ router.get('/:courseId/', (req, res, next) => {
...
+					total: 0,
+					data: [],
+				};
+			}));
 	}
 
 	// ############################ end requests to new Editor ##########################
+	const [
+		course,
+		_lessons,
```
Comment (tofixx) [inline controllers/courses.js]: lint: no underscore?

File: controllers/courses.js [hunk 5]
```diff
@@ -620,99 +622,122 @@ router.get('/:courseId/', (req, res, next) => {
...
-				url: `/courses/${req.params.courseId}/topics/${lesson._id}/`,
-			}));
-
-			const homeworks = (_homeworks.data || []).map((assignment) => {
-				assignment.url = `/homework/${assignment._id}`;
-				return assignment;
-			});
+	try {
+		// ############################## check if new Editor options should show #################
+		const edtrUser = (res.locals.currentUser.features || []).includes('edtr');
```
Comment (tofixx) [inline controllers/courses.js]: use more descriptive name like `currentUserHasFeatureEditor`

File: controllers/courses.js [hunk 6]
```diff
@@ -620,99 +622,122 @@ router.get('/:courseId/', (req, res, next) => {
...
+		const userHasEditorEnabled = EDITOR_URL && edtrUser;
```
Comment (tofixx) [inline controllers/courses.js]: the meaning is not correct? why doing this if feature is globally disabled?

File: controllers/courses.js [hunk 7]
```diff
@@ -620,99 +622,122 @@ router.get('/:courseId/', (req, res, next) => {
...
-					),
-					// #################### new Editor, till replacing old one ######################
-					newLessons,
-					isNewEdtrioActivated,
-					scopedCoursePermission: scopedPermissions[res.locals.currentUser._id],
-				}),
-			);
-		})
-		.catch(next);
+		const baseUrl = req.headers.origin || HOST || 'http://localhost:3100';
```
Comment (tofixx) [inline controllers/courses.js]: move this (`HOST || 'http://localhost:3100'`) to config/globals.js or use HOST only

File: controllers/courses.js (hunk 1 above)
Comment (CeEv) [reply controllers/courses.js]: I do not add this envs, only remove it from request callback into global scope.

File: controllers/courses.js (hunk 2 above)
Comment (CeEv) [reply controllers/courses.js]: The same i do not add it. It is the same like host and server urls. This is not a feature. The new editor is a base MS in future. But the client should work without, like calendar and content MS.

File: controllers/courses.js (hunk 7 above)
Comment (CeEv) [reply controllers/courses.js]: I do not add it and i do not know for what origin in context of CSRF is used. It is adding by aron.

File: controllers/courses.js (hunk 4 above)
Comment (CeEv) [reply controllers/courses.js]: It is old code too. In this PR i add only editorBackendIsAlive check and resort the error handling.

File: controllers/courses.js (hunk 5 above)
Comment (CeEv) [reply controllers/courses.js]: currentUser is the send user information from server and this includes features as flag. The model has the flag features.

File: controllers/courses.js (hunk 6 above)
Comment (CeEv) [reply controllers/courses.js]: Some closed beta tester can have the flag to use the new editor. If the editor backend is set and alive and a visible editor lesson exist, it should display it to the students.

**Pull request merged**